main.py

```diff
@@ -28,80 +28,115 @@
+    sample = nusc.get("sample", sample_token)
     result = {}
     cam_data = []
     for nuscenes_sensor, argoverse_sensor in SENSOR_NAMES.items():
-        sensor_data = nusc.get('sample_data', sample['data'][nuscenes_sensor])
-        calibration = nusc.get('calibrated_sensor', sensor_data['calibrated_sensor_token'])
+        sensor_data = nusc.get("sample_data", sample["data"][nuscenes_sensor])
+        calibration = nusc.get(
+            "calibrated_sensor", sensor_data["calibrated_sensor_token"]
+        )
         transformation_dict = {}
-        transformation_dict['rotation'] = {'coefficients': calibration['rotation']}
-        transformation_dict['translation'] = calibration['translation']
+        transformation_dict["rotation"] = {"coefficients": calibration["rotation"]}
+        transformation_dict["translation"] = calibration["translation"]
         if is_camera_sensor(nuscenes_sensor):
             camera_info = {}
-            camera_info['key'] = "image_raw_" + argoverse_sensor
+            camera_info["key"] = "image_raw_" + argoverse_sensor
             value = {}
-            value["focal_length_x_px_"] = calibration['camera_intrinsic'][0][0]
-            value["focal_length_y_px_"] = calibration['camera_intrinsic'][1][1]
-            value["focal_center_x_px_"] = calibration['camera_intrinsic'][0][2]
-            value["focal_center_y_px_"] = calibration['camera_intrinsic'][1][2]
-            value["skew_"] = calibration['camera_intrinsic'][0][1]
+            value["focal_length_x_px_"] = calibration["camera_intrinsic"][0][0]
+            value["focal_length_y_px_"] = calibration["camera_intrinsic"][1][1]
+            value["focal_center_x_px_"] = calibration["camera_intrinsic"][0][2]
+            value["focal_center_y_px_"] = calibration["camera_intrinsic"][1][2]
+            value["skew_"] = calibration["camera_intrinsic"][0][1]
             # Nuscenes does not provide distortion coefficients.
             value["distortion_coefficients_"] = [0, 0, 0]
-            value['vehicle_SE3_camera_'] = transformation_dict
-            camera_info['value'] = value
+            value["vehicle_SE3_camera_"] = transformation_dict
+            camera_info["value"] = value
             cam_data.append(camera_info)
         else:
             # Nuscenes has one lidar sensor.
             result["vehicle_SE3_down_lidar_"] = transformation_dict
             result["vehicle_SE3_up_lidar_"] = transformation_dict
-    result['camera_data_'] = cam_data
+    result["camera_data_"] = cam_data
     return result
 
+
 def round_to_micros(t_nanos: int, base: int = 1000) -> int:
     """
     Round nanosecond timestamp to nearest microsecond timestamp
     """
     return base * round(t_nanos / base)
 
+
+def write_ply(
+    points_egovehicle: np.ndarray,
+    points: np.ndarray,
+    output_sensor_path: str,
+    timestamp: int,
+) -> None:
+    """
+    Write the ply files corresponding to pointcloud.
```

Comment:
Thanks for all the updates @JeevanjotS . Looks really good. One small comment here -- do you mind adding  in the docstring the shape of `points_egovehicle` and of `points` and what sensor frame `points` are in?

Reply:
maybe:
```python
def write_ply(
    points_egovehicle: np.ndarray,
    points_lidar: np.ndarray,
    output_sensor_path: str,
    timestamp: int,
) -> None:
    """
    Write the ply files corresponding to pointcloud.

    Args:
          points_egovehicle Nx3 in egovehicle frame
          points_lidar: Nx5 in lidar frame
          ...
```

Reply:
Done

Reply:
thanks